qiita_db/data.py

```diff
@@ -1257,6 +1257,60 @@ def create(cls, processed_params_table, processed_params_id, filepaths,
         pd.add_filepaths(filepaths, conn_handler)
         return cls(pd_id)
 
+    @classmethod
+    def delete(cls, processed_data_id):
+        """Removes the processed data with id processed_data_id
+
+        Parameters
+        ----------
+        processed_data_id : int
+            The processed data id
+
+        Raises
+        ------
+        QiitaDBUnknownIDError
+            If the processed data id doesn't exist
+        QiitaDBStatusError
+            If the processed data status is not sandbox
+        """
+        if cls(processed_data_id).status != 'sandbox':
+            raise QiitaDBStatusError(
+                "Illegal operation on non sandbox processed data")
+
+        conn_handler = SQLConnectionHandler()
+
+        analysis_ids = [str(_id[0]) for _id in conn_handler.execute_fetchall(
+            "SELECT DISTINCT analysis_id FROM qiita.analysis_sample WHERE "
+            "processed_data_id = {0} ORDER BY "
+            "analysis_id".format(processed_data_id))]
+
+        if analysis_ids:
+            raise QiitaDBError(
+                "Processed data %d is linked to (meta)analyses: %s" %
```

Comment:
One thing I just thought of if showing names: this will potentially show names of analyses that the user does not have access to. Probably not the end of the world, but could be a privacy issue.